src/main/java/org/opensearch/neuralsearch/search/collector/HybridTopScoreDocCollector.java

```diff
@@ -108,12 +108,17 @@ public void collect(int doc) throws IOException {
                 }
                 // Increment total hit count which represents unique doc found on the shard
                 totalHits++;
+                hitsThresholdChecker.incrementHitCount();
```

Comment:
this is a gap team noted while working on the fix, not directly related to reported issue